core/domain/suggestion_services.py

```diff
@@ -0,0 +1,367 @@
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_type(
+            suggestion_type)]
+
+
+def get_suggestions_by_target_id(target_type, target_id):
+    """Gets a list of suggestions to the entity with the given ID.
+
+    Args:
+        target_type: str. The type of target entity the suggestion is linked to.
+        target_id: str. The ID of the target entity the suggestion is linked to.
+
+    Returns:
+        list(Suggestions): a list of suggestions linked to the entity.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_target_id(
+            target_type, target_id)]
+
+
+def update_suggestion(suggestion):
+    """Updates the given sugesstion. The properties that can be edited are the
+    following:
+        status, reviewer_id, assigned_reviewer_id, payload.
+
+    Args:
+        suggesstion: Suggestion. The domain object of the suggestion to be
+            updated.
+    """
+
+    suggestion_model = suggestion_models.SuggestionModel.get_by_id(
+        suggestion.suggestion_id)
+
+    if suggestion_model.status == suggestion_models.STATUS_IN_REVIEW:
+        suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
+        suggestion_model.assigned_reviewer_id = suggestion.assigned_reviewer_id
+
+        suggestion_model.change_cmd = suggestion.change_cmd
+
+    suggestion_model.status = suggestion.status
+
+    suggestion_model.put()
+
+
+def get_thread_id_from_suggestion_id(suggestion_id):
+    """Gets the thread_id from the suggestion_id.
+
+    Args:
+        suggestion_id: str. The ID of the suggestion.
+
+    Returns:
+        str. The thread ID linked to the suggestion.
+    """
+
+    # Temporarily. Once the feedback migration is complete, this function will
+    # be removed.
+    return suggestion_id[suggestion_id.rfind('.') + 1:]
+
+
+def is_suggestion_valid(suggestion, reviewer_id):
+    """Validates a suggestion. This function should be called before accepting
+    the suggestion.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
+            validated.
+        reviewer_id: str. The ID of the reviewer who is reviewing the
+            suggestion. If the suggestion is invalid, the thread status will be
+            set to 'ignored' by the reviewer.
+
+    Returns:
+        bool. The validity of the suggestion.
+    """
+
+    if (
+            suggestion.suggestion_type ==
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT):
+        states = exp_services.get_exploration_by_id(
+            suggestion.target_id).states
+        state = suggestion.change_cmd['state_name']
+        if state not in states:
+            suggestion.status = suggestion_models.STATUS_INVALID
+            suggestion.final_reviewer_id = reviewer_id
+            suggestion.assigned_reviewer_id = ''
```

Comment:
Do not use '' to represent None. Ditto elsewhere.

Reply:
Done